### PR TITLE
Adding service to begin watering a program

### PR DESCRIPTION
**Pull request by disforw**
This PR will add a service “start_program” that requires a switch.program entity. Once called it will start watering that program.

### CODE REVIEW

File: custom_components/bhyve/switch.py [hunk 1]
```diff
@@ -275,6 +280,30 @@ async def async_turn_on(self, **kwargs: Any) -> None:
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the switch off."""
         await self._set_state(False)
+        
+    async def start_program(self):
+        """Begins running a program."""
+        station_payload = self._program["program"]
+        await self._send_program_message(station_payload)
```
Comment (sebr) [inline custom_components/bhyve/switch.py:287]:
```suggestion
        program_payload = self._program["program"]
        await self._send_program_message(program_payload)
```

File: custom_components/bhyve/switch.py [hunk 2]
```diff
@@ -275,6 +280,30 @@ async def async_turn_on(self, **kwargs: Any) -> None:
...
+            iso_time = now.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+            payload = {
+                "event": EVENT_CHANGE_MODE,
+                "mode": "manual",
+                "device_id": self._device_id,
+                "timestamp": iso_time,
+                "program": station_payload,
+            }
+            _LOGGER.info(payload)
```
Comment (sebr) [inline custom_components/bhyve/switch.py:301]:
```suggestion
            _LOGGER.debug(payload)
```

Don't log the payload to `info`. You can log this to debug instead.

File: custom_components/bhyve/switch.py [hunk 3]
```diff
@@ -275,6 +280,30 @@ async def async_turn_on(self, **kwargs: Any) -> None:
...
+        await self._send_program_message(station_payload)
+
+    async def _send_program_message(self, station_payload):
+        try:
+            now = datetime.datetime.now()
+            iso_time = now.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+            payload = {
+                "event": EVENT_CHANGE_MODE,
+                "mode": "manual",
```
Comment (sebr) [inline custom_components/bhyve/switch.py:296]: It looks like these were copied from `_send_station_message`. Are these attributes required on this event? They smell funny to me... why would a program have a mode?

File: custom_components/bhyve/switch.py [hunk 4]
```diff
@@ -608,7 +637,7 @@ async def start_watering(self, minutes):
         station_payload = [{"station": self._zone_id, "run_time": minutes}]
         self._is_on = True
         await self._send_station_message(station_payload)
-
+
```
Comment (sebr) [inline custom_components/bhyve/switch.py:640]:
```suggestion

```

**Review by sebr — CHANGES_REQUESTED** [review]
Thanks for the PR - a few small questions and changes requested before I can merge.

File: custom_components/bhyve/switch.py (hunk 3 above)
Comment (disforw) [reply custom_components/bhyve/switch.py:296]: They seem to be required, take a look at my findings that I pasted in issue #145 . I monitored the WS and that's what came across when running a program.

File: custom_components/bhyve/switch.py [hunk 5]
```diff
@@ -608,7 +637,6 @@ async def start_watering(self, minutes):
         station_payload = [{"station": self._zone_id, "run_time": minutes}]
         self._is_on = True
         await self._send_station_message(station_payload)
-
```
Comment (sebr) [inline custom_components/bhyve/switch.py:611]: Oops can you please revert this line removal

**Review by sebr — APPROVED** [review]
Code looks good, I will update `Readme.md` with updated docs

**Pull request merged**